### PR TITLE
PixelShaderGen: apply zfreeze before ztextures

### PR DESCRIPTION
**Pull request by degasus**
The zfreeze option freezes the depth plane of the rasterization.
So this is done before the TEV stages, where the z-textures are applied.

### CODE REVIEW

**Comment by JMC47** [comment]
This completely breaks zfreeze for me.

**Comment by Sonicadvance1** [comment]
Were all the issues fixed with the buildbot spouting errors about earlier revisions of this PR?

**Comment by degasus** [comment]
@Sonicadvance1  Yes, I've fixed the broken cast.

File: Source/Core/VideoCommon/PixelShaderGen.cpp [hunk 1]
```diff
@@ -557,18 +556,30 @@ static inline void GeneratePixelShader(T& out, DSTALPHA_MODE dstAlphaMode, API_T
...
 
-	// FastDepth means to trust the depth generated in perspective division.
-	// It should be correct, but it seems not to be as accurate as required. TODO: Find out why!
-	// For disabled FastDepth we just calculate the depth value again.
-	// The performance impact of this additional calculation doesn't matter, but it prevents
-	// the host GPU driver from performing any early depth test optimizations.
-	if (g_ActiveConfig.bFastDepthCalc)
-		out.Write("\tint zCoord = int(rawpos.z * 16777216.0);\n");
-	else
+	out.Write("\tint zCoord = int(rawpos.z * 16777216.0);\n");
```
Comment (Armada651) [inline Source/Core/VideoCommon/PixelShaderGen.cpp]: Why use fast-depth for zfreeze instead of the slow-depth?

**Comment by neobrain** [comment]
Why?

**Comment by degasus** [comment]
@neobrain I'm pretty sure zfreeze is applied at the rasterization. But the z-texture is done at the TEV stages, so afterwards. Did you try this on hardware?

**Comment by neobrain** [comment]
@degasus No, I just think it's better to provide descriptions and reasonings for non-obvious changes so that other people can understand what's happening.

**Comment by degasus** [comment]
@Armada651 I've fixed the coding style issues. Is it goo now?

**Comment by Armada651** [comment]
Yeah, LGTM

**Pull request merged**